loops/c/code.c

```diff
@@ -3,14 +3,17 @@
 #include "stdint.h"
 #include "time.h"
 
+// A casey inspired macro
+#define REM(a, b) (int)(((double)a) - ((double)b)*trunc(((double)a) / ((double)b)))
+
 int main (int argc, char** argv) {
   int u = atoi(argv[1]);               // Get an input number from the command line
   srand(time(NULL));                   // FIX random seed
   int r = rand() % 10000;              // Get a random integer 0 <= r < 10k
   int32_t a[10000] = {0};              // Array of 10k elements initialized to 0
   for (int i = 0; i < 10000; i++) {    // 10k outer loop iterations
     for (int j = 0; j < 100000; j++) { // 100k inner loop iterations, per outer loop iteration
-      a[i] = a[i] + j%u;               // Simple sum
+      a[i] = a[i] + REM(j,u)               // Simple sum
```

Comment:
COOL, but why the compiler does not do

Reply:
I don't know to be honest, the only time I saw a compiler optimizes modules is if you tried doing module power of 2, in that case it replaces it with "and" instruction, so for example if 

a = 35
b = 32 

a % b = a & (b - 1)  

which in this case is 3